src/test/unit/helperFunctions/test_process.py

```diff
@@ -1,14 +1,18 @@
 import logging
+from multiprocessing import Event
 from time import sleep
 
 import pytest
+from flaky import flaky
 
 from helperFunctions.process import ExceptionSafeProcess, check_worker_exceptions, new_worker_was_started
 
+check_exceptions_finished = Event()
+
 
 def breaking_process(wait: bool = False):
     if wait:
-        sleep(0.5)
+        check_exceptions_finished.wait(timeout=5)
```

Comment:
Maybe we could use adjustable timeouts through environment variables?

Reply:
Good idea but it is probably best to do this for all sleep/wait times in the tests at once (and that is probably out of scope for this PR)